assets/src/models/createDetourMachine.ts

```diff
@@ -96,15 +96,25 @@ export const createDetourMachine = setup({
       {
         routePatternId?: RoutePatternId
         startPoint?: ShapePoint
+        waypoints: ShapePoint[]
         endPoint?: ShapePoint
       }
-    >(async ({ input: { routePatternId, startPoint, endPoint } }) => {
-      if (routePatternId && startPoint && endPoint) {
-        return fetchFinishedDetour(routePatternId, startPoint, endPoint)
-      } else {
-        throw "Missing finished detour inputs"
+    >(
+      async ({
+        input: { routePatternId, startPoint, waypoints, endPoint },
+      }) => {
+        if (routePatternId && startPoint && endPoint) {
+          return fetchFinishedDetour(
+            routePatternId,
+            startPoint,
+            waypoints,
+            endPoint
+          )
+        } else {
+          throw "Missing finished detour inputs"
+        }
```

Comment:
suggestion(non-blocking): It'd be nice to flatten this while were here
```suggestion
        if (!(routePatternId && startPoint && endPoint)) {
          throw "Missing finished detour inputs"
        }
        return fetchFinishedDetour(
          routePatternId,
          startPoint,
          waypoints,
          endPoint
        )
```